lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2461,6 +2461,15 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
       }
     }
 
+    // Reject variable if it is a stored property with a structurally
+    // uninhabited type
+    if (VD->isInstanceMember() && VD->hasStorage() && 
```

Comment:
I don't think this is limited to members; any stored type properties or globals or even locals would have the same problem.